black_it/loss_functions/moment_networks/moment_network.py

```diff
@@ -0,0 +1,147 @@
+# Black-box ABM Calibration Kit (Black-it)
+# Copyright for this module (C) 2023 Jonathan Chassot and Banca d'Italia
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+This module contains a Moment Network implementation.
+
+- Chassot, J. (2023).
+  Paper name
+
+Algorithm description:
+
+"""
+
+
+from typing import Callable, Tuple
+
+import numpy as np
+import torch
+import torch.nn as nn
+from numpy.typing import NDArray
+
+from black_it.samplers.base import BaseSampler
+from black_it.search_space import SearchSpace
+
+MSE_LOSS = nn.MSELoss()
+
+
+# TODO: Some issues with the parameters imho, the optimizer as a Callback is not
```

Comment:
an idea to solve it from ChatGPT:

https://chat.openai.com/share/43b7067f-9a6c-47a5-b2a2-9bf64b0ebd8e